android/vpnClient/src/main/java/org/mozilla/firefox/vpn/qt/VPNActivity.java

```diff
@@ -201,4 +201,10 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
     super.onActivityResult(requestCode, resultCode, data);
   }
+
+  @Override
+  protected void onDestroy() {
+    unbindService(mConnection);
+    super.onDestroy();
+  }
```

Comment:
How come this is relate to the Qt update?

Reply:
We have been leaking this, however due to QT failing to recreate the activity in <6.6 we never noticed. Now we would see that.